src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -391,83 +401,119 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
                                                                                       instanceId + "'"));
 
         // Retrieve all resources attached to the instance
-        NetworkInterface networkInterface = vm.getPrimaryNetworkInterface();
-        com.microsoft.azure.management.network.Network network = networkInterface.primaryIpConfiguration().getNetwork();
-        NetworkSecurityGroup networkSecurityGroup = networkInterface.getNetworkSecurityGroup();
-        Optional<PublicIpAddress> optionalPublicIPAddress = Optional.ofNullable(vm.getPrimaryPublicIpAddress());
+        List<NetworkInterface> networkInterfaces = vm.networkInterfaceIds()
+                                                     .stream()
+                                                     .map(id -> azureService.networkInterfaces().getById(id))
+                                                     .collect(Collectors.toList());
+        List<com.microsoft.azure.management.network.Network> networks = networkInterfaces.stream()
+                                                                                         .flatMap(networkInterface -> networkInterface.ipConfigurations()
+                                                                                                                                      .values()
+                                                                                                                                      .stream())
+                                                                                         .map(NicIpConfigurationBase::getNetwork)
+                                                                                         .filter(network -> Optional.ofNullable(network)
+                                                                                                                    .isPresent())
+                                                                                         .distinct()
+                                                                                         .collect(Collectors.toList());
+        List<NetworkSecurityGroup> networkSecurityGroups = networkInterfaces.stream()
+                                                                            .map(NetworkInterfaceBase::getNetworkSecurityGroup)
+                                                                            .filter(networkSecurityGroup -> Optional.ofNullable(networkSecurityGroup)
```

Comment:
Good to know, thx for the hint! done in 74910f08d17a230ac31a2f9eb7ef663013913eec